### PR TITLE
Align with new ReleaseCycle and App CR changes

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/5277

* update k8s dependencies to `kubernetes-1.13.2`
* update vendor
* fix metricsresource.WrapConfig

* watch ReleaseCycle CR
* align App CR changes
* drop version bundle

The goal here is to get rid of the old Release CR and introducing the new ReleaseCycle CR.

### CODE REVIEW

File: service/controller/v1/resource/app/desired.go [hunk 1]
```diff
@@ -4,15 +4,20 @@ import (
...
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	releaseCR, err := key.ToCustomResource(obj)
+	releaseCycleCR, err := key.ToReleaseCycleCR(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	releaseCR, err := r.g8sClient.ReleaseV1alpha1().Releases(releaseCycleCR.GetNamespace()).Get(releaseCycleCR.Spec.Release.Name, metav1.GetOptions{})
```
Comment (TheoBrigitte) [inline service/controller/v1/resource/app/desired.go]: This is far from the logic we want to have here. I will update it in later PR.

File: service/controller/v1/resource/chartconfig/current.go [hunk 2]
```diff
@@ -30,7 +30,12 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	customResource, err := key.ToCustomResource(obj)
+	releaseCycleCR, err := key.ToReleaseCycleCR(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	releaseCR, err := r.g8sClient.ReleaseV1alpha1().Releases(releaseCycleCR.GetNamespace()).Get(releaseCycleCR.Spec.Release.Name, metav1.GetOptions{})
```
Comment (TheoBrigitte) [inline service/controller/v1/resource/chartconfig/current.go]: This is far from the logic we want to have here. I will update it in later PR.

File: service/controller/v1/resource/chartconfig/desired.go [hunk 3]
```diff
@@ -12,17 +13,22 @@ import (
...
+	releaseCycleCR, err := key.ToReleaseCycleCR(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	var desiredChartConfigCRs []*v1alpha1.ChartConfig
+	releaseCR, err := r.g8sClient.ReleaseV1alpha1().Releases(releaseCycleCR.GetNamespace()).Get(releaseCycleCR.Spec.Release.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```
Comment (TheoBrigitte) [inline service/controller/v1/resource/chartconfig/desired.go]: This is far from the logic we want to have here. I will update it in later PR.

File: service/controller/v1/resource_set.go [hunk 4]
```diff
@@ -126,9 +125,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 	}
 
 	{
-		c := metricsresource.WrapConfig{
-			Name: config.ProjectName,
```
Comment (TheoBrigitte) [inline service/controller/v1/resource_set.go]: Updated according to : https://github.com/giantswarm/operatorkit/pull/276

File: service/controller/v1/resource_set.go [hunk 5]
```diff
@@ -144,24 +141,11 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
...
-			return false
-		}
-
-		if key.VersionBundleVersion(customResource) == VersionBundle().Version {
-			return true
-		}
-
-		return false
-	}
-
```
Comment (TheoBrigitte) [inline service/controller/v1/resource_set.go]: We are moving from version bundle to label selector. Dropping this for now and will add label selector in later PR.

File: service/controller/v1/key/key.go [hunk 6]
```diff
@@ -20,20 +20,16 @@ const (
...
+		return releasev1.ReleaseCycle{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", &releasev1.ReleaseCycle{}, v)
 	}
 	customResource := *customResourcePointer
 
 	return customResource, nil
 }
-
-func VersionBundleVersion(customResource v1alpha1.Release) string {
-	return customResource.Spec.VersionBundle.Version
-}
```
Comment (TheoBrigitte) [inline service/controller/v1/key/key.go]: We are moving from version bundle to label selector.

File: service/controller/v1/resource/chartconfig/desired.go [hunk 7]
```diff
@@ -3,7 +3,8 @@ package chartconfig
 import (
 	"context"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	corev1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	releasev1 "github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
```
Comment (xh3b4sd) [inline service/controller/v1/resource/chartconfig/desired.go]: The alias does not express the truth. When we want to prefix, it should be `releasev1alpha1`. The important path is not `.../release/v1`.

File: service/controller/v1/resource_set.go (hunk 5 above)
Comment (xh3b4sd) [reply service/controller/v1/resource_set.go]: How does the deployment work though? I feel like dropping this here should be the last step to activate the new behaviour.

Comment (TheoBrigitte) [reply service/controller/v1/resource_set.go]: This operator is not deployed yet.

File: service/controller/v1/resource/chartconfig/desired.go (hunk 7 above)
Comment (TheoBrigitte) [reply service/controller/v1/resource/chartconfig/desired.go]: Good point.

Renamed all:
applicationv1 > applicationv1alpha1
corev1 > corev1alpha1
releasev1 > releasev1alpha1

**Pull request merged**